src/libraries/System.Collections.Concurrent/tests/ConcurrentDictionary/ConcurrentDictionary.Generic.Tests.cs

```diff
@@ -70,16 +71,22 @@ private static List<string> GenerateCollidingStrings(int count)
             static Func<string, int> GetHashCodeFunc(ConcurrentDictionary<string, string> cd)
             {
                 // If the layout of ConcurrentDictionary changes, this will need to change as well.
-                object tables = cd.GetType().GetField("_tables", BindingFlags.Instance | BindingFlags.NonPublic).GetValue(cd);
-                Assert.NotNull(tables);
 
-                FieldInfo comparerField = tables.GetType().GetField("_comparer", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public);
-                Assert.NotNull(comparerField);
+                FieldInfo tablesField = AssertNotNull(typeof(ConcurrentDictionary<string, string>).GetField("_tables", BindingFlags.Instance | BindingFlags.NonPublic));
+                Type tablesType = AssertNotNull(Type.GetType("System.Collections.Concurrent.ConcurrentDictionary`2+Tables, System.Collections.Concurrent"));
```

Comment:
```suggestion
                Type tablesType = Type.GetType("System.Collections.Concurrent.ConcurrentDictionary`2+Tables, System.Collections.Concurrent", throwOnError: true);;
```
I expect that this is necessary to make trimmer data flow to see that the field preserve the `Tables::_comparer` field needs to be reflectable.

`AssertNotNull` wrapper strips the information about type identify and prevents the `GetField("_comparer"`, ...` from being treated as intrinsic that makes the field reflectable.

Reply:
Good catch - yes, this would break the data flow analysis.

Reply:
To remove some of the mystery of whether dataflow analysis will work or not, it's possible to temporarily enable `<EnableTrimAnalyzer>true</EnableTrimAnalyzer>` in the test project - it will flag things if they cannot be understood.

It would be nice if we could eventually enable it for all tests.